packages/driver/src/multi-domain/domain_fn.ts

```diff
@@ -77,8 +77,10 @@ export const handleOriginFn = (Cypress: Cypress.Cypress, cy: $Cy) => {
   }
 
   const setRunnableStateToPassed = () => {
-    // TODO: We're telling the runnable that it has passed to avoid a timeout on the last (empty) command. Normally this would be set inherently by running (runnable.run) the test.
-    // Set this to passed regardless of the state of the test, the runnable isn't responsible for reporting success.
+    // HACK: We're telling the runnable that it has passed to avoid a timeout
+    // on the last (empty) command. Normally this would be set inherently by
+    // running runnable.run() the test. Set this to passed regardless of the
+    // state of the test, the runnable isn't responsible for reporting success.
```

Comment:
what is responsible here?

Reply:
The runnable is the test itself, so it's passed or failed by everything in the test passing or something failing, be it a command or assertion or an error being thrown.

The runnable in the secondary driver is mostly stubbed out and its state doesn't influence the actual runnable in the primary driver.

It exists at all in the secondary driver because various parts of the driver rely on it having certain properties and methods. Ideally, it wouldn't need to exist in the secondary driver at all, but removing that would potentially mean a fairly broad refactor of the driver. For the sake of completing the experimental release and reducing the broader impact of the cross-origin support changes, we've not done that as of yet.

We [plan to refactor and optimize the code that goes into the secondary driver](https://github.com/cypress-io/cypress/issues/19888), so we might be able to refactor out the need for the runnable in the secondary driver as part of that effort.